From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>

Add a MonadFail instance to Client
This is needed for some examples to compile

diff -urN jupyter-0.9.0-orig/src/Jupyter/Client.hs /home/kinetik/tmp/jupyter-0.9.0/src/Jupyter/Client.hs
--- jupyter-0.9.0-orig/src/Jupyter/Client.hs	2016-08-13 17:12:56.000000000 -0600
+++ /home/kinetik/tmp/jupyter-0.9.0/src/Jupyter/Client.hs	2021-08-15 11:25:28.948659268 -0600
@@ -145,7 +145,8 @@
 -- Use 'sendClientRequest' and 'sendClientComm' to construct 'Client' values, the 'Monad' interface to
 -- manipulate them, and 'runClient' to supply all needed connection info and run the action.
 newtype Client a = Client { unClient :: ReaderT ClientState IO a }
-  deriving (Functor, Applicative, Monad, MonadIO, MonadReader ClientState, MonadThrow, MonadCatch, MonadMask)
+  deriving ( Functor, Applicative, Monad, MonadIO, MonadReader ClientState
+           , MonadThrow, MonadCatch, MonadMask, MonadFail)
 
 -- | A connection to a kernel from a client.
 --
